warehouse/bigquery.go

```diff
@@ -157,32 +158,80 @@ func (bq *BigQuery) LoadToWarehouse(storageRef string, startTime time.Time) erro
 	return bq.waitForJob(job)
 }
 
-// EnsureCompatibleExportTable ensures that the all the fields present in the hauser schema are present in the BigQuery table schema
-// If the table exists, it compares the schema of the export table in BigQuery to the schema in hauser and adds any missing fields
-// if the table doesn't exist, it creates a new export table with all the fields specified in hauser
-func (bq *BigQuery) EnsureCompatibleExportTable() error {
-	// Get Hauser Schema
-	// this is required if we create a new table or if we have to compare to the existing table schema
-	hauserSchema, err := bigquery.InferSchema(BundleEvent{})
+func convertSchema(s Schema, existing bigquery.Schema) (bigquery.Schema, error) {
+	bqs := make([]*bigquery.FieldSchema, len(s))
+	for i, field := range s {
+		// Not checking ok here because we may not need it
+		bqType, ok := bigQueryTypeMap[field.FieldType]
+
+		if i < len(existing) {
+			// Make sure the existing schema is compatible with the provided hauser schema
+			if !strings.EqualFold(field.DBName, existing[i].Name) {
+				return nil, fmt.Errorf(
+					"columns names don't match at index %d: expected %s, got %s", i, existing[i].Name, field.DBName)
+			} else if field.FullStoryFieldName != "" && bqType != existing[i].Type {
```

Comment:
When would `FullStoryFieldName` be empty? Shouldn't it always be present?

Reply:
Oh I guess it's that fallthrough case for `GetFieldForName`? Why do we want this inconsistency though? If we need it, can we doc when it is missing?

Reply:
Yeah, we need it because the `Schema` contains all the columns in the database. This can include columns that are not part of an export. The non-empty `FullStoryFieldName`s are used to build the export request.